src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1344,9 +1354,66 @@ private static bool IsLessDerivedThanAny<TMember>(int index, TypeSymbol type, Ar
                 {
                     return true;
                 }
+                else if (currentType.IsExtension)
+                {
+                    if (currentType.ExtendedTypeNoUseSiteDiagnostics is { } extendedType)
+                    {
+                        if (extendedType.Equals(type, TypeCompareKind.ConsiderEverything))
+                        {
+                            return true;
+                        }
+
+                        if ((extendedType.IsClassType() || extendedType.IsEnumType() || extendedType.IsStructType() || extendedType.IsDelegateType())
```

Comment:
>(extendedType.IsClassType() || extendedType.IsEnumType() || extendedType.IsStructType() || extendedType.IsDelegateType()) [](http://example.com/codeflow?start=28&length=121)

Can we replace this with ```!extendedType.IsInterface```?

Reply:
Perhaps we also need to filter out type parameters.

Reply:
It might be worth rearranging the `if`s. First check type parameters, then interfaces, then the rest.